source/slang/slang-ir-clone.cpp

```diff
@@ -51,16 +51,16 @@ IRInst* cloneInstAndOperands(IRCloneEnv* env, IRBuilder* builder, IRInst* oldIns
     SLANG_ASSERT(builder);
     SLANG_ASSERT(oldInst);
 
-    // This logic will not handle any instructions
-    // with special-case data attached, but that only
-    // applies to `IRConstant`s at this point, and those
-    // should only appear at the global scope rather than
-    // in function bodies.
-    //
-    // TODO: It would be easy enough to extend this logic
-    // to handle constants gracefully, if it ever comes up.
-    //
-    SLANG_ASSERT(!as<IRConstant>(oldInst));
+    // IRConstants contain data other than just the operands, so cloning them
+    // is done separately.
+    IRConstant* oldConstant = as<IRConstant>(oldInst);
+    if (oldConstant)
+    {
+        IRConstant* newConstant = builder->getClonedConstantValue(*oldConstant);
```

Comment:
This will always create a new inst without going through the deduplication logic. If we already have an inst that represents the cloned inst, this will break the invariant that all constants in global scope are deduplicated. 

Given that the case will only ever arise for pointer literals because their types may be generic, we should handle just the PtrLit case and leave other opcodes as is, and then call emitPtrLit instead of doing all these cloning.

Reply:
Okay, sounds good. This would also keep the PR much smaller & cleaner since `getClonedConstantValue` would no longer need to separated from `_findOrEmitConstant`.